src/pages/api/token/pr/[id].ts

```diff
@@ -0,0 +1,44 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getPayentRequestById, getPayentRequestByIdIncludeToken } from '@/lib/paymentRequestModels';
+import { AuthenticatedRequest, CheckPaymentRequestResponse } from '@/types';
+import { runAuthMiddleware } from '@/utils/middleware';
+
+export default async function handler(req: AuthenticatedRequest, res: NextApiResponse) {
+   await runAuthMiddleware(req, res);
```

Comment:
there are a lot of these basic checks repeated in every api handler. I have some ideas how we can abstract that to avoid writing the same checks everywhere. not important for this PR though